otherlibs/dune-rpc/private/dune_rpc_private.ml

```diff
@@ -692,6 +692,7 @@ module Client = struct
               ~menu
           in
           let* () = Fiber.Ivar.fill handler_var handler in
+          client.handler_initialized <- true;
```

Comment:
Perhaps this should be called before the fill? Otherwise the waiters on handler_var will be able to observe `handler_initialized = false`.

Reply:
Good catch, fixed.